src/drivers/ulanding/ulanding.cpp

```diff
@@ -0,0 +1,392 @@
+
+	virtual int 			init();
+
+	int				start();
+
+private:
+	bool				_task_should_exit;
+	int 				_task_handle;
+	char 				_port[20];
+	int				_class_instance;
+	int				_orb_class_instance;
+	orb_advert_t			_distance_sensor_topic;
+
+	static void task_main_trampoline(int argc, char *argv[]);
+	void task_main();
+};
+
+namespace radar
+{
+Radar	*g_dev;
+}
+
+Radar::Radar(const char *port) :
+	CDev("Radar", RANGE_FINDER0_DEVICE_PATH),
+	_task_should_exit(false),
+	_task_handle(-1),
+	_class_instance(-1),
+	_orb_class_instance(-1),
+	_distance_sensor_topic(nullptr)
+{
+	/* store port name */
+	strncpy(_port, port, sizeof(_port));
+	/* enforce null termination */
+	_port[sizeof(_port) - 1] = '\0';
+
+	// disable debug() calls
+	_debug_enabled = false;
+}
+
+Radar::~Radar()
+{
+
+	if (_class_instance != -1) {
+		unregister_class_devname(RANGE_FINDER_BASE_DEVICE_PATH, _class_instance);
+	}
+
+	if (_task_handle != -1) {
+		/* task wakes up every 100ms or so at the longest */
+		_task_should_exit = true;
+
+		/* wait for a second for the task to quit at our request */
+		unsigned i = 0;
+
+		do {
+			/* wait 20ms */
+			usleep(20000);
+
+			/* if we have given up, kill it */
+			if (++i > 50) {
+				px4_task_delete(_task_handle);
+				break;
+			}
+		} while (_task_handle != -1);
+	}
+
+	if (_class_instance != -1) {
+		unregister_class_devname(RANGE_FINDER_BASE_DEVICE_PATH, _class_instance);
+	}
+
+	orb_unadvertise(_distance_sensor_topic);
+}
+
+int
+Radar::init()
+{
+	/* status */
+	int ret = 0;
+
+	do { /* create a scope to handle exit conditions using break */
+
+		/* do regular cdev init */
+		ret = CDev::init();
+
+		if (ret != OK) {
+			PX4_WARN("cdev init failed");
+			break;
+		}
+
+		int fd = px4_open(RANGE_FINDER0_DEVICE_PATH, 0);
+
+		if (fd < 0) {
+			PX4_WARN("failed to open range finder device");
+			ret = 1;
+			break;
+		}
+
+		px4_close(fd);
+
+		/* open fd */
+		fd = px4_open(_port, O_RDWR | O_NOCTTY);
+
+		if (fd < 0) {
+			PX4_WARN("failed to open serial device");
+			ret = 1;
+			break;
+		}
+
+		struct termios uart_config;
+
+		int termios_state;
+
+		/* fill the struct for the new configuration */
+		tcgetattr(fd, &uart_config);
+
+		/* clear ONLCR flag (which appends a CR for every LF) */
+		uart_config.c_oflag &= ~ONLCR;
+
+		/* no parity, one stop bit */
+		uart_config.c_cflag	 &= ~(CSTOPB | PARENB);
+
+		unsigned speed = B115200;
+
+		/* set baud rate */
+		if ((termios_state = cfsetispeed(&uart_config, speed)) < 0) {
+			PX4_WARN("ERR CFG: %d ISPD", termios_state);
+			ret = 1;
+			break;
+		}
+
+		if ((termios_state = cfsetospeed(&uart_config, speed)) < 0) {
+			PX4_WARN("ERR CFG: %d OSPD\n", termios_state);
+			ret = 1;
+			break;
+		}
+
+		if ((termios_state = tcsetattr(fd, TCSANOW, &uart_config)) < 0) {
+			PX4_WARN("ERR baud %d ATTR", termios_state);
+			ret = 1;
+			break;
+		}
+
+		px4_close(fd);
+
+		_class_instance = register_class_devname(RANGE_FINDER_BASE_DEVICE_PATH);
+
+		struct distance_sensor_s ds_report = {};
```

Comment:
It would be better to send the sensor ID and everything here. This is an all-zero report which will make the system think it has two different sensors, out of which one sends only a single measurement.